### ISSUE TITLE
Unknown error creating VM if disk-size is not valid

### ISSUE THREAD

**Issue by jorgemoralespou** [minishift/minishift#1068]
With virtualbox:

````
$ minishift config set disk-size 40960
No Minishift instance exists. New disk-size setting will be applied on next 'minishift start'

$ minishift start
Starting local OpenShift cluster using 'virtualbox' hypervisor...
E0627 13:46:56.160140    1462 start.go:275] Error starting the VM: Error creating the VM. Error creating machine: Error in driver during machine creation: exit status 1. Retrying.
E0627 13:46:56.210722    1462 start.go:275] Error starting the VM: Error getting the state for host: machine does not exist. Retrying.
E0627 13:46:56.256466    1462 start.go:275] Error starting the VM: Error getting the state for host: machine does not exist. Retrying.
Error starting the VM: Error creating the VM. Error creating machine: Error in driver during machine creation: exit status 1
Error getting the state for host: machine does not exist
Error getting the state for host: machine does not exist

$ minishift config set disk-size 40g
You currently have an existing Minishift instance. Changes to the disk-size setting are only applied when a new Minishift instance is created.
To let the configuration changes take effect, you must delete the current instance with 'minishift delete' and then start a new one with 'minishift start'.

$ minishift delete
Deleting the Minishift VM...
Minishift VM deleted.

$ minishift config view
- cpus                 : 2
- disk-size            : 40g
- image-caching        : true
- memory               : 8192
- vm-driver            : virtualbox

$ minishift start
Starting local OpenShift cluster using 'virtualbox' hypervisor...
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ... OK
-- Checking for existing OpenShift container ... OK
-- Checking for openshift/origin:v1.5.1 image ...
   Pulling image openshift/origin:v1.5.1
   Pulled 0/3 layers, 3% complete
   Pulled 1/3 layers, 87% complete
   Pulled 2/3 layers, 94% complete
   Pulled 3/3 layers, 100% complete
   Extracting
   Image pull complete
-- Checking Docker daemon configuration ... OK
-- Checking for available ports ... OK
-- Checking type of volume mount ...
   Using Docker shared volumes for OpenShift volumes
-- Creating host directories ... OK
 ....
````

**Comment by LalatenduMohanty** [comment]
I think this can be added to pre-flight check i.e. if the disk size is a valid size and it is possible on the current host.

**Comment by hferentschik** [comment]
I think it is partly indeed partly related to #179.

**Comment by hferentschik** [comment]
I think it is also a regression introduced here https://github.com/minishift/minishift/issues/486

**Comment by gbraad** [comment]
---
Todo
---

  * start options
    * [x] treat string with only numbers as megabytes
    * [x] treat GB as gigabytes, MB as megabytes, etc
    * [x] treat G as gigabytes, M as megabytes 
  * parse config options
    * [ ] treat set options as mentioned above

```
$ minishift start --disk-size 10G
-- Minishift VM will be configured with ...
   Memory:    2 GB
   vCPUs :    2
   Disk size: 10 GB
-- Starting Minishift VM ... \ ^C
$ minishift start --disk-size 10GB
-- Minishift VM will be configured with ...
   Memory:    2 GB
   vCPUs :    2
   Disk size: 10 GB
-- Starting Minishift VM ... \ ^C
$ minishift start --disk-size 10000
-- Minishift VM will be configured with ...
   Memory:    2 GB
   vCPUs :    2
   Disk size: 10 GB
-- Starting Minishift VM ... / ^C
```

**Comment by gbraad** [comment]
Will also look at `--memory`

**Comment by hferentschik** [comment]
Fixed via pull request #1271

**Comment by gbraad** [comment]
Memory will be done as #1149
